Support int for kAbs, half for fmod/remainder in SimpleIREvaluator

penguinwu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#49357 Support int for kAbs, half for fmod/remainder in SimpleIREvaluator**

Summary: This is a follow-up fix for PR #48679, where the previous PR
adds support for integer inputs to aten::abs by promoting integers to
float and then demote the result back to integers. This PR supports
integer inputs to aten::abs more efficiently in the SimpleIREvaluator
by allowing implementing integer inputs for kAbs (renamed from kFabs).
- Rename kFabs to kAbs
- Add support for integer input to kAbs in SimpleIREvalator (note that:
llvm_codegen and cuda_codegen already supports integer inputs to kAbs)
- Support half type for `fmod` and `remainder` in `kernel.cpp`

Test Plan:
- `PYTORCH_TENSOREXPR_DONT_USE_LLVM=1 python test/test_jit_fuser_te.py
TestTEFuser.test_unary_ops`
- `python test/test_jit_fuser_te.py TestTEFuser.test_unary_ops`

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25545791](https://our.internmc.facebook.com/intern/diff/D25545791)

Commit:
Support integral types for kAbs in SimpleIREvaluator

Summary: This is a follow-up fix for PR #48679, where the previous PR
adds support for integer inputs to aten::abs by promoting integers to
float and then demote the result back to integers. This PR supports
integer inputs to aten::abs more efficiently in the SimpleIREvaluator
by allowing implementing integer inputs for kAbs (renamed from kFabs).
- Rename kFabs to kAbs
- Add support for integer input to kAbs in SimpleIREvalator (note that:
llvm_codegen and cuda_codegen already supports integer inputs to kAbs)

Test Plan:
- `PYTORCH_TENSOREXPR_DONT_USE_LLVM=1 python test/test_jit_fuser_te.py
TestTEFuser.test_unary_ops`
- `python test/test_jit_fuser_te.py TestTEFuser.test_unary_ops`

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

Commit:
Update on "Support integral types for kAbs in SimpleIREvaluator"


Summary: This is a follow-up fix for PR #48679, where the previous PR
adds support for integer inputs to aten::abs by promoting integers to
float and then demote the result back to integers. This PR supports
integer inputs to aten::abs more efficiently in the SimpleIREvaluator
by allowing implementing integer inputs for kAbs (renamed from kFabs).
- Rename kFabs to kAbs
- Add support for integer input to kAbs in SimpleIREvalator (note that:
llvm_codegen and cuda_codegen already supports integer inputs to kAbs)

Test Plan:
- `PYTORCH_TENSOREXPR_DONT_USE_LLVM=1 python test/test_jit_fuser_te.py
TestTEFuser.test_unary_ops`
- `python test/test_jit_fuser_te.py TestTEFuser.test_unary_ops`

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

eellison commented on torch/csrc/jit/tensorexpr/eval.h:
```diff
@@ -946,7 +949,8 @@ class SimpleIREvaluator : public CodeGen, public IRVisitor {
         return std::lgamma(v);
       case kFrac:
-        T intpart;
+        using X = std::conditional_t<std::is_integral<T>::value, float, T>;
```
nice! i couldnt figure this out myself

eellison reviewed (approved):
Nice! LGTM!

Commit:
Update on "Support integral types for kAbs in SimpleIREvaluator"


Summary: This is a follow-up fix for PR #48679, where the previous PR
adds support for integer inputs to aten::abs by promoting integers to
float and then demote the result back to integers. This PR supports
integer inputs to aten::abs more efficiently in the SimpleIREvaluator
by allowing implementing integer inputs for kAbs (renamed from kFabs).
- Rename kFabs to kAbs
- Add support for integer input to kAbs in SimpleIREvalator (note that:
llvm_codegen and cuda_codegen already supports integer inputs to kAbs)

Test Plan:
- `PYTORCH_TENSOREXPR_DONT_USE_LLVM=1 python test/test_jit_fuser_te.py
TestTEFuser.test_unary_ops`
- `python test/test_jit_fuser_te.py TestTEFuser.test_unary_ops`

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25545791](https://our.internmc.facebook.com/intern/diff/D25545791)

[ghstack-poisoned]

Commit:
Update on "Support integral types for kAbs in SimpleIREvaluator"


Summary: This is a follow-up fix for PR #48679, where the previous PR
adds support for integer inputs to aten::abs by promoting integers to
float and then demote the result back to integers. This PR supports
integer inputs to aten::abs more efficiently in the SimpleIREvaluator
by allowing implementing integer inputs for kAbs (renamed from kFabs).
- Rename kFabs to kAbs
- Add support for integer input to kAbs in SimpleIREvalator (note that:
llvm_codegen and cuda_codegen already supports integer inputs to kAbs)

Test Plan:
- `PYTORCH_TENSOREXPR_DONT_USE_LLVM=1 python test/test_jit_fuser_te.py
TestTEFuser.test_unary_ops`
- `python test/test_jit_fuser_te.py TestTEFuser.test_unary_ops`

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25545791](https://our.internmc.facebook.com/intern/diff/D25545791)

[ghstack-poisoned]

Commit:
Update on "Support integral types for kAbs in SimpleIREvaluator"


Summary: This is a follow-up fix for PR #48679, where the previous PR
adds support for integer inputs to aten::abs by promoting integers to
float and then demote the result back to integers. This PR supports
integer inputs to aten::abs more efficiently in the SimpleIREvaluator
by allowing implementing integer inputs for kAbs (renamed from kFabs).
- Rename kFabs to kAbs
- Add support for integer input to kAbs in SimpleIREvalator (note that:
llvm_codegen and cuda_codegen already supports integer inputs to kAbs)

Test Plan:
- `PYTORCH_TENSOREXPR_DONT_USE_LLVM=1 python test/test_jit_fuser_te.py
TestTEFuser.test_unary_ops`
- `python test/test_jit_fuser_te.py TestTEFuser.test_unary_ops`

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25545791](https://our.internmc.facebook.com/intern/diff/D25545791)

[ghstack-poisoned]

Commit:
Update on "Support integral types for kAbs in SimpleIREvaluator"


Summary: This is a follow-up fix for PR #48679, where the previous PR
adds support for integer inputs to aten::abs by promoting integers to
float and then demote the result back to integers. This PR supports
integer inputs to aten::abs more efficiently in the SimpleIREvaluator
by allowing implementing integer inputs for kAbs (renamed from kFabs).
- Rename kFabs to kAbs
- Add support for integer input to kAbs in SimpleIREvalator (note that:
llvm_codegen and cuda_codegen already supports integer inputs to kAbs)

Test Plan:
- `PYTORCH_TENSOREXPR_DONT_USE_LLVM=1 python test/test_jit_fuser_te.py
TestTEFuser.test_unary_ops`
- `python test/test_jit_fuser_te.py TestTEFuser.test_unary_ops`

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25545791](https://our.internmc.facebook.com/intern/diff/D25545791)

[ghstack-poisoned]

Commit:
Update on "Support int for kAbs, half for fmod/remainder in SimpleIREvaluator"


Summary: This is a follow-up fix for PR #48679, where the previous PR
adds support for integer inputs to aten::abs by promoting integers to
float and then demote the result back to integers. This PR supports
integer inputs to aten::abs more efficiently in the SimpleIREvaluator
by allowing implementing integer inputs for kAbs (renamed from kFabs).
- Rename kFabs to kAbs
- Add support for integer input to kAbs in SimpleIREvalator (note that:
llvm_codegen and cuda_codegen already supports integer inputs to kAbs)
- Support half type for `fmod` and `remainder` in `kernel.cpp`

Test Plan:
- `PYTORCH_TENSOREXPR_DONT_USE_LLVM=1 python test/test_jit_fuser_te.py
TestTEFuser.test_unary_ops`
- `python test/test_jit_fuser_te.py TestTEFuser.test_unary_ops`

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25545791](https://our.internmc.facebook.com/intern/diff/D25545791)

[ghstack-poisoned]

Commit:
Update on "Support int for kAbs, half for fmod/remainder in SimpleIREvaluator"


Summary: This is a follow-up fix for PR #48679, where the previous PR
adds support for integer inputs to aten::abs by promoting integers to
float and then demote the result back to integers. This PR supports
integer inputs to aten::abs more efficiently in the SimpleIREvaluator
by allowing implementing integer inputs for kAbs (renamed from kFabs).
- Rename kFabs to kAbs
- Add support for integer input to kAbs in SimpleIREvalator (note that:
llvm_codegen and cuda_codegen already supports integer inputs to kAbs)
- Support half type for `fmod` and `remainder` in `kernel.cpp`

Test Plan:
- `PYTORCH_TENSOREXPR_DONT_USE_LLVM=1 python test/test_jit_fuser_te.py
TestTEFuser.test_unary_ops`
- `python test/test_jit_fuser_te.py TestTEFuser.test_unary_ops`

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25545791](https://our.internmc.facebook.com/intern/diff/D25545791)

[ghstack-poisoned]

Commit:
Update on "Support int for kAbs, half for fmod/remainder in SimpleIREvaluator"


Summary: This is a follow-up fix for PR #48679, where the previous PR
adds support for integer inputs to aten::abs by promoting integers to
float and then demote the result back to integers. This PR supports
integer inputs to aten::abs more efficiently in the SimpleIREvaluator
by allowing implementing integer inputs for kAbs (renamed from kFabs).
- Rename kFabs to kAbs
- Add support for integer input to kAbs in SimpleIREvalator (note that:
llvm_codegen and cuda_codegen already supports integer inputs to kAbs)
- Support half type for `fmod` and `remainder` in `kernel.cpp`

Test Plan:
- `PYTORCH_TENSOREXPR_DONT_USE_LLVM=1 python test/test_jit_fuser_te.py
TestTEFuser.test_unary_ops`
- `python test/test_jit_fuser_te.py TestTEFuser.test_unary_ops`

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25545791](https://our.internmc.facebook.com/intern/diff/D25545791)

[ghstack-poisoned]

Commit:
Update on "Support int for kAbs, half for fmod/remainder in SimpleIREvaluator"


Summary: This is a follow-up fix for PR #48679, where the previous PR
adds support for integer inputs to aten::abs by promoting integers to
float and then demote the result back to integers. This PR supports
integer inputs to aten::abs more efficiently in the SimpleIREvaluator
by allowing implementing integer inputs for kAbs (renamed from kFabs).
- Rename kFabs to kAbs
- Add support for integer input to kAbs in SimpleIREvalator (note that:
llvm_codegen and cuda_codegen already supports integer inputs to kAbs)
- Support half type for `fmod` and `remainder` in `kernel.cpp`

Test Plan:
- `PYTORCH_TENSOREXPR_DONT_USE_LLVM=1 python test/test_jit_fuser_te.py
TestTEFuser.test_unary_ops`
- `python test/test_jit_fuser_te.py TestTEFuser.test_unary_ops`

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25545791](https://our.internmc.facebook.com/intern/diff/D25545791)

[ghstack-poisoned]
